synapseclient/core/download/download_functions.py

```diff
@@ -161,6 +170,105 @@ async def download_file_entity(
     entity.cacheDir = os.path.dirname(download_path)
 
 
+async def download_file_entity_model(
+    *,
+    download_location: str,
+    file: "File",
+    if_collision: str,
+    submission: str,
+    synapse_client: Optional["Synapse"] = None,
+) -> None:
+    """
+    Download file entity
+
+    Arguments:
+        download_location: The download location
```

Comment:
Would suggest making the docstrings more detailed since they also serve as documentation for users. e.g "The location on your local machine where the entity will be downloaded. Note that this location will be searched for any previous versions of the entity that were downloaded already, and will be handled with the `if_collision` argument"

Reply:
So for these functions within `core` they are not meant to be used directly by users of the python client. They're functions to be used by higher-level code. There is a disclaimer on the core python docs page that reads:

https://python-docs.synapse.org/reference/core/

> This section is for super users / developers only. These functions are subject to change as they are internal development functions. Use at your own risk.

For example `entity_factory.py` uses this, and that function has more in-depth docstrings.

With that being said: I'll update the docstring with your example since it's great!

Reply:
```
        download_location: The location on disk where the entity will be downloaded. If
            there is a matching file at the location, the download collision will be
            handled according to the `if_collision` parameter.
```

Reply:
Added in this commit: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1111/commits/c269dc7d830d437d15d2c382f6cc10d251ba3bb9

Reply:
Excellent thanks!